[DOCS] Update wildcard query documentation (#120251)
elastic/elasticsearch: docs/reference/query-dsl/wildcard-query.asciidoc

[[wildcard-query-notes]]
==== Notes

> Wildcard queries can be resource-intensive, particularly when using leading wildcards (*). To improve performance, minimize their use and consider alternatives like <<analysis-ngram-tokenizer, n-gram tokenizer>>, which allow for more efficient searching, though they can increase index size. For better accuracy and performance, combine wildcard queries with other query types (e.g., match or bool) to narrow results first. 

Reviewer: ```suggestion
Wildcard queries using `*` can be resource-intensive, particularly with leading wildcards. To improve performance, minimize their use and consider alternatives like the <<analysis-ngram-tokenizer,n-gram tokenizer>>. While this allows for more efficient searching, it may increase index size. For better performance and accuracy, combine wildcard queries with other query types like <<query-dsl-match-query,`match`>> or <<query-dsl-bool-query,`bool`>> to first narrow down results.
```

- Tries to avoid parentheses as much as possible.
- Keeps sentences simple
- Adds helpful links

Author: Looks good, thanks @leemthompo . This was a small change so I just came straight from docs. I now see that when I press edit, it goes to the specific version rather than main. I can just change the URL and go from there
